napari/layers/points/points.py

```diff
@@ -1098,16 +1101,27 @@ def mode(self, mode):
             return
         old_mode = self._mode
 
+        if old_mode == Mode.ADD:
+            self.mouse_drag_callbacks.remove(add)
+        elif old_mode == Mode.SELECT:
+            # add mouse drag and move callbacks
+            self.mouse_drag_callbacks.remove(select)
```

Comment:
It might be nice to add a test that goes from `Mode.ADD` to `Mode.SELECT` (and vice versa) to make sure the callbacks are properly added/removed. I think the tests only go from `Mode.PAN_ZOOM` to `Mode.ADD` or `Mode.SELECT`.

Reply:
Good suggestion - two tests have been added, one for add and one for select. Thanks!